packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,193 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
     };
   }
 }
+
+/**
+ * The type returned from the `bind()` method in {@link HttpGatewayRoutePathMatch}.
+ */
+export interface HttpGatewayRoutePathMatchConfig {
+  /**
+   * Gateway route configuration for matching on the complete URL path of the request.
+   *
+   * @default - no matching will be performed on the complete URL path
+   */
+  readonly wholePathMatch?: CfnGatewayRoute.HttpPathMatchProperty;
+
+  /**
+   * Gateway route configuration for matching on the prefix of the URL path of the request.
+   *
+   * @default - no matching will be performed on the prefix of the URL path
+   */
+  readonly prefixPathMatch?: string;
+
+  /**
+   * Gateway route configuration for rewriting the complete URL path of the request..
+   *
+   * @default - no rewrite will be performed on the request's complete URL path
+   */
+  readonly wholePathRewrite?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * Gateway route configuration for rewriting the prefix of the URL path of the request.
+   *
+   * @default - rewrites the request's URL path to '/'
+   */
+  readonly prefixPathRewrite?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines HTTP gateway route matching based on the URL path of the request.
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   * If path rewrite is specified, the provided `path must also end with the `/` character.
+   *
+   * @param path the exact path to match on
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static exactly(path: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
```

Comment:
This should be:

```suggestion
  public static exactly(path: string, rewriteTo: string | undefined = '/'): HttpGatewayRoutePathMatch {
```

And please explain the docs that you need to pass `null` / `undefined` to `rewriteTo` if you want to turn off rewriting.

Reply:
I confirmed that when whole path match is specified (*in other words, everything except prefix path match), there will be **no automatic rewrite on the path.**

Reply:
> I understand that the default if you don't specify the `CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty` is equivalent to `{ exact: '/' }` (the entire matched path path gets rewritten to /).

Sorry, I was very wrong with this statement. While matching on whole path, if you don't specify the `CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty` is NOT equivalent to `{ exact: '/' }`

Reply:
Ok, that changes things substantially then 🙂.

Before I comment on the PR, one "meta" question.

Currently, the defaults for rewrites are very different in `HttpGatewayRoutePathMatch` between `startsWith()` and `exactly()` / `regex()`.

@Seiya6329 do you consider this a problem? Should we try to unify them, or are you fine with these differences between the different path match options?

Reply:
> Currently, the defaults for rewrites are very different in `HttpGatewayRoutePathMatch` between `startsWith()` and `exactly()` / `regex()`.

I think only difference between those two groups is **whether there is a default rewrite or not**. We can improve the JSdoc and/or `README` doc if it seems not clear for an user.

> do you consider this a problem? Should we try to unify them, or are you fine with these differences between the different path match options?

Based on above comment, I think current design is acceptable. We can still design `exactly()` and `regex()` to behave similarly by setting the default to `/` and handling `''`.
However, this behavior can be **misleading for an user since App mesh API does not have a concept for default path rewrite.**

Reply:
Another caveat can be if App Mesh actually introduces a concept of default whole path rewrite that rewrites to something different than `/` (*although I strong doubt it).

Reply:
> Another caveat can be if App Mesh actually introduces a concept of default whole path rewrite that rewrites to something different than `/` (*although I strongly doubt it).

That would be a breaking change to the service, which seems unlikely.